src/read.c

```diff
@@ -2629,11 +2611,7 @@ UInt ReadEvalFile(Obj *evalResult)
     STATE(Prompt) = SyQuiet ? "" : "> ";
 
     /* remember the old reader context                                     */
-    stackNams   = rs->StackNams;
-    readTop     = rs->ReadTop;
-    readTilde   = rs->ReadTilde;
     tilde       = STATE(Tilde);
-    currLHSGVar = rs->CurrLHSGVar;
```

Comment:
Real change: since we freshly allocated the `ReaderState` on the stack, there is no need to "preserve" its "previous" content.